packages/indicators/src/__tests__/builders/buildArithmetic/buildArithmetic.test.ts

```diff
@@ -0,0 +1,149 @@
+      /B.* has no aggregation defined/,
+    );
+  });
+
+  it('calls `aggregator.fetchAnalytics` with `fetchOptions`', async () => {
+    const config = {
+      formula: 'A + B',
+      aggregation: { A: ['MOST_RECENT'], B: ['MOST_RECENT'] },
+    };
+    const fetchOptions = { organisationUnitCodes: ['TO'] };
+    await buildArithmetic({ aggregator, config, fetchOptions });
+
+    expect(aggregator.fetchAnalytics).toHaveBeenCalledOnceWith(
+      expect.anything(),
+      fetchOptions,
+      expect.anything(),
+    );
+  });
+
+  describe('calculations for a single orgUnit/period combo', () => {
+    const testData: [string, ArithmeticConfig, AnalyticValue[]][] = [
+      [
+        'simple calculation - integer',
+        {
+          formula: 'A_TO_2019 + B_TO_2019',
+          aggregation: aggregationsByCode(['A_TO_2019', 'B_TO_2019']),
+        },
+        [{ organisationUnit: 'TO', period: '2019', value: 3 }],
+      ],
+      [
+        'simple calculation - float',
+        {
+          formula: 'A_TO_2019 / B_TO_2019',
+          aggregation: aggregationsByCode(['A_TO_2019', 'B_TO_2019']),
+        },
+        [{ organisationUnit: 'TO', period: '2019', value: 0.5 }],
+      ],
+      [
+        'complex calculation',
+        {
+          formula: '((A_TO_2019 + B_TO_2019) * C_TO_2019) / (D_TO_2019 - E_TO_2019)',
+          aggregation: aggregationsByCode([
+            'A_TO_2019',
+            'B_TO_2019',
+            'C_TO_2019',
+            'D_TO_2019',
+            'E_TO_2019',
+          ]),
+        },
+        [{ organisationUnit: 'TO', period: '2019', value: -9 }],
+      ],
+      [
+        'division with zero',
+        {
+          formula: 'A_TO_2019 / (A_TO_2019 + B_TO_2019 - C_TO_2019)',
+          aggregation: aggregationsByCode(['A_TO_2019', 'B_TO_2019', 'C_TO_2019']),
+        },
+        [{ organisationUnit: 'TO', period: '2019', value: Infinity }],
+      ],
+      [
+        'some data elements are undefined in the orgUnit/period combo',
+        {
+          formula: 'A_TO_2020 + UNDEFINED_TO_2019',
+          aggregation: aggregationsByCode(['A_TO_2020', 'UNDEFINED_TO_2019']),
+        },
+        [],
+      ],
+    ];
+
+    it.each(testData)('%s', (_, config, expected) =>
+      expect(
+        buildArithmetic({ aggregator, config, fetchOptions: {} }),
+      ).resolves.toIncludeSameMembers(expected),
+    );
+  });
+
+  describe('calculations for multiple orgUnit/period combos', () => {
+    const testData: [string, ArithmeticConfig, AnalyticValue[]][] = [
+      [
+        'all data elements are defined in all combos',
+        {
+          formula: 'A_TOPG_201920 + B_TOPG_201920',
+          aggregation: aggregationsByCode(['A_TOPG_201920', 'B_TOPG_201920']),
+        },
+        [
+          { organisationUnit: 'TO', period: '2019', value: 3 },
+          { organisationUnit: 'TO', period: '2020', value: 30 },
+          { organisationUnit: 'PG', period: '2019', value: 3.3000000000000003 },
```

Comment:
Oh I love JavaScript maths...

```javascript
1.1 + 2.2 === 3.3000000000000003 // true
1.1 + 2.2 === 3.3 // false!!!
```

🤦

Reply:
Haha yeah crazy - this is actually not specific to js!

![Screen Shot 2020-09-01 at 11 23 13 AM](https://user-images.githubusercontent.com/1274422/91778052-967e9c80-ec45-11ea-87a5-822011dbf571.png)

Reply:
Haha nice, I take that back 😆